### PR TITLE
[browser] Do not reset model upon a tab close. Fixes JB#55345 OMP#JOLLA-333

### PR DESCRIPTION
**Pull request by rainemak**

### CODE REVIEW

**Review by mkenttala — APPROVED** [review]

File: apps/browser/qml/pages/components/TabGridView.qml [hunk 1]
```diff
@@ -123,14 +123,4 @@ SilicaGridView {
     }
 
     onLoadedChanged: positionViewAtIndex(model.activeTabIndex, GridView.Center)
-
-    Connections {
```
Comment (chriadam) [inline apps/browser/qml/pages/components/TabGridView.qml:127]: Why was this done in the past?  Why is it no longer required?  Neither the bug nor the commit message have any useful information about this..

Comment (rainemak) [reply apps/browser/qml/pages/components/TabGridView.qml:127]: We missed this in the review when changes were landed to the tab model. This causes horrible flickering in the tab model when you have multiple tabs open and you close one as the model resets when a tab is closed. If there are errors like model data not being updated correctly, those should be fixed instead.

**Comment by rainemak** [comment]
Amended commit message and bug description.

**Comment by chriadam** [comment]
Thanks, LGTM

**Review by chriadam — APPROVED** [review]

**Review by okodron — APPROVED** [review]

**Pull request merged**